machine/minit.c

```diff
@@ -78,7 +78,8 @@ static void delegate_traps()
     (1U << CAUSE_BREAKPOINT) |
     (1U << CAUSE_LOAD_PAGE_FAULT) |
     (1U << CAUSE_STORE_PAGE_FAULT) |
-    (1U << CAUSE_USER_ECALL);
+    (1U << CAUSE_USER_ECALL) |
+    (1U << CAUSE_SOFTWARE_CHECK_FAULT);
```

Comment:
What if `CAUSE_SOFTWARE_CHECK_FAULT` is not supported (never happening) in the current implementation?

In the next lines, there is an assertion check for `assert(read_csr(medeleg) == exceptions);`. This would require the `CAUSE_SOFTWARE_CHECK_FAULT` bit in medeleg to be writable. If the corresponding bit is not writable, pk will abort. Should pk have some configurable options to disable this ISA extension?

Reply:
Yes, this is a problem, but I don't think configurability of pk is the best solution.  Simply masking off these optional codes from the assertion is the answer.  The fix is here: 4f3debe4d04f56d31089c1c716a27e2d5245e9a1